src/ios/SentryCordova.m

```diff
@@ -8,121 +8,245 @@
+      SentryEnvelopeHeader *envelopeHeader =
+          [[SentryEnvelopeHeader alloc] initWithId:eventId andSdkInfo:sdkInfo];
+
+      NSError *error;
+      NSData *envelopeItemData =
+          [NSJSONSerialization dataWithJSONObject:envelopeDict[@"payload"]
+                                          options:0
+                                            error:&error];
+      if (nil != error) {
+        result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                     messageAsBool:NO];
+      } else {
+        NSString *itemType = envelopeDict[@"payload"][@"type"];
+        if (itemType == nil) {
+          // Default to event type.
+          itemType = @"event";
         }
-    }];
+
+        SentryEnvelopeItemHeader *envelopeItemHeader =
+            [[SentryEnvelopeItemHeader alloc]
+                initWithType:itemType
+                      length:envelopeItemData.length];
+        SentryEnvelopeItem *envelopeItem =
+            [[SentryEnvelopeItem alloc] initWithHeader:envelopeItemHeader
+                                                  data:envelopeItemData];
+
+        SentryEnvelope *envelope =
+            [[SentryEnvelope alloc] initWithHeader:envelopeHeader
+                                        singleItem:envelopeItem];
+
+#if DEBUG
+        [[SentrySDK currentHub] captureEnvelope:envelope];
+#else
+                if ([envelopeDict[@"payload"][@"level"] isEqualToString:@"fatal"]) {
+                    // Storing to disk happens asynchronously with captureEnvelope
+                    // We need to make sure the event is written to disk before resolving the promise.
+                    // This could be replaced by SentrySDK.flush() when available.
+                    [[[SentrySDK currentHub] getClient] storeEnvelope:envelope];
+                } else {
+                    [[SentrySDK currentHub] captureEnvelope:envelope];
+                }
+#endif
+      }
+    } else {
+      result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                   messageAsBool:NO];
+    }
+
+    [self.commandDelegate sendPluginResult:result
+                                callbackId:command.callbackId];
+  }];
 }
 
 - (void)addBreadcrumb:(CDVInvokedUrlCommand *)command {
-    [self.commandDelegate runInBackground:^{
-        NSDictionary *jsonBreadcrumb = [command.arguments objectAtIndex:0];
-        if ([jsonBreadcrumb isKindOfClass:NSDictionary.class]) {
-          SentryBreadcrumb *breadcrumb = [SentryJavaScriptBridgeHelper createSentryBreadcrumbFromJavaScriptBreadcrumb:jsonBreadcrumb];
-          [SentryClient.sharedClient.breadcrumbs addBreadcrumb:breadcrumb];
-          CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:[breadcrumb serialize]];
-          [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
-        }
+  [self.commandDelegate runInBackground:^{
```

Comment:
Do we need to run this one the background?

Reply:
Hmm now that I think of it none of the scope sync methods really need to run in the background anymore as they're not copying the whole scope each time anymore.